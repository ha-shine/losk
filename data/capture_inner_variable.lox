fun outer() {
    var x = "outside";
    fun inner() {
        print x;
    }

    inner();
}

outer();

fun modifyUpvalue() {
    var name = "Shine";
    print "Hello, " + name;

    fun setName(value) {
        name = value;
    }

    setName("John");
    print "Hello, " + name;
}

modifyUpvalue();
