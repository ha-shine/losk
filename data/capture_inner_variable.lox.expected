outside
Hello, Shine
Hello, John
